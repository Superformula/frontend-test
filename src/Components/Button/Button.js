import React from 'react';
import PropTypes from 'prop-types';
import "./Button.scss";

function Button(props) {
  return (
    <button className={props.size}>
      {props.children}
    </button>
  );
}

Button.propTypes = {
  size: PropTypes.string
};

export default Button;