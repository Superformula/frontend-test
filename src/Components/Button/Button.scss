// import for color variables
@import '../../styles/index';

.button {
  @include font-regular;
  background-color: $midnightblue;
  border-color: $midnightblue;
  border-radius: 2px;
  color: $white;
  font-size: 14px;
  letter-spacing: 1px;
  line-height: 16px;
  max-width: 420px;
  padding: 13px 0;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
  
  &.inverted {
    background-color: transparent;
    border: 1px solid $midnightblue;
    color: $midnightblue;
  }
}

