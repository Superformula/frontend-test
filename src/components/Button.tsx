import React, { PropsWithChildren } from "react";
import styled from "@emotion/styled";

type ButtonProps = PropsWithChildren<{
  variant?: "primary" | "secondary";
}>;

export function Button({ variant = "primary", children }: ButtonProps) {
  return <Wrapper variant={variant}>{children}</Wrapper>;
}

const Wrapper = styled("button")<Pick<ButtonProps, "variant">>`
  border: none;
  border-radius: 2px;
  padding: 16px 8px;
  box-sizing: border-box;
  text-align: center;
  letter-spacing: 0.857143px;
  text-transform: uppercase;
  font-size: 12px;
  font-weight: 500;
  color: ${props => (props.variant === "primary" ? "#fff" : "#C8C8C8")};
  border: ${props => props.variant === "secondary" && "1px solid #E6E6E6;"};
  background-color: ${props =>
    props.variant === "primary" ? "#002B56" : "#fff"};
`;
