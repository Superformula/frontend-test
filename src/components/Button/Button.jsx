import PropTypes from 'prop-types';
import styled from 'styled-components';
import { variant } from 'styled-system';

const buttonStyle = variant({
  key: 'buttons',
});

const Button = styled.button`
  ${buttonStyle}
  display: inline-flex;
  justify-content: center;
  width: ${props => props.block ? '100%' : 'auto'}
  padding: 12px 24px;
  outline: 0;
  font-size: ${props => props.small ? props.theme.fontSizes[0] : props.theme.fontSizes[1]};
  font-stretch:normal;
  text-transform: uppercase;
  border-radius: 2px;
  transition: all cubic-bezier(0.4, 0.0, 0.2, 1) 0.25s;
  cursor: pointer;
  &:disabled {
    cursor: default;
  }
  &:hover:not(:disabled) {
    transform: translateY(2px);
  }
`;

Button.defaultProps = {
  secondary: false,
  block: false,
};

Button.propTypes = {
  secondary: PropTypes.bool,
  block: PropTypes.bool,
};

export default Button;
