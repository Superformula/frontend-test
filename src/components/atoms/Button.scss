.button {
    color: #021F44;
    border-color: #021F44;
    font-size: 10px;
    font-weight: 500; 
    min-height: 30px;
    min-width: 125px;
    cursor: pointer;

    &.large {
        min-width: 340px;
        min-height: 40px;
        font-size: 11px;
    }

    &:hover {
        color: white;
        background-color: #021F44;
    }

    &:active {
        color: white;
        background-color:slategray;
    }

    &:focus {
        outline:0;
    }

    &:disabled {
        border: 1px solid #999999;
        background-color: #cccccc;
        color: #666666;
    }
}