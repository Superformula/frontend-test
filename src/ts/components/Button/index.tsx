// Imports
// -------

// Libraries
import React from 'react';
// Style
import style from './style.module.css';

// Internal
// --------

export interface ButtonProps {
    text: string,
    tier?: 'primary' | 'secondary',
    size?: 'small' | 'large' | 'full',
    disabled?: boolean,
    onClick?(): any
}

const Button: React.FC<ButtonProps> = ({text, tier= 'primary', size= 'small', disabled= false, onClick= () => {}}) => {
    //
    // Render

    const className = [style.base, style[tier], style[size], disabled ? style.disabled : ''].join(' ');

    return (
        <button className={ className } onClick={ onClick }>{ text }</button>
    );
}

// Exports
// -------

export default Button;
