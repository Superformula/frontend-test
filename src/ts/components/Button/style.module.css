.base {
    font-size: 14px;
    line-height: 16px;
    font-weight: normal;
    border-radius: 2px;
    height: 48px;
    letter-spacing: 1px;
    text-transform: uppercase;
    border: none;
    cursor: pointer;
    transition: all .2s ease-in;
    position: relative;
}

.primary {
    color: #fff;
    background-color: #002b56;
}

.secondary {
    color: #002b56;
    background-color: #fff;
    border: #002b56 1px solid;
    font-weight: bold;
}

.small {
    height: 38px;
    width: 151px;
}

.large {
    width: 416px;
}

.full {
    width: 100%;
}

.disabled {
    background-color: #fff;
    border: #e6e6e6 1px solid;
    color: #c8c8c8;
    cursor: unset;
    font-weight: bold;
}
