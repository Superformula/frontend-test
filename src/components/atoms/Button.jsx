import React from 'react';
import './Button.scss';
import { PropTypes } from 'prop-types';
import combineClasses from '../../util/CombineClasses';

const Button = (props) => {
    const buttonClass = combineClasses([
        'button',
        props.large && 'large',
        props.className
    ]);

    return (
        <button
            className={ buttonClass }
            onClick={ props.onClick }
        >
            { props.text }
        </button>
    );
}

Button.propTypes = {
    text: PropTypes.string,
    className: PropTypes.string,
    onClick: PropTypes.func,
    large: PropTypes.bool,
};

export default Button;