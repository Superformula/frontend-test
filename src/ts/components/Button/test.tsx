// Imports
// -------

// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import Button from './index';

// Internal
// --------

const defaultProps = {
  text: 'Testing'
};

const getComponent = (props = {}) => renderer.create(
    <Button {...defaultProps} {...props} />,
);

// Tests
// -----

describe('src/ts/components/Button', () => {
    let component;

    describe('Rendering', () => {
        it('renders default correctly', () => {
            component = getComponent();
            expect(component).toMatchSnapshot();
        });

        describe('tier', () => {
            it('renders "secondary" correctly', () => {
                component = getComponent({ tier: 'secondary'});
                expect(component).toMatchSnapshot();
            });
        });

        describe('size', () => {
            it('renders "large" correctly', () => {
                component = getComponent({ size: 'large'});
                expect(component).toMatchSnapshot();
            });
        });

        describe('disabled', () => {
            it('renders true correctly', () => {
                component = getComponent({ disabled: true });
                expect(component).toMatchSnapshot();
            });
        });
    });

    describe('Interaction', () => {
        it('calls "onClick" correctly', () => {
            const mockFn = jest.fn();
            component = getComponent({ onClick: mockFn });
            component.root.findByType('button').props.onClick();
            expect(mockFn).toHaveBeenCalled();
        });
    });
});
