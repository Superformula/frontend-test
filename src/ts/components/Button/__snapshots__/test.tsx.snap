// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/ts/components/Button Rendering disabled renders true correctly 1`] = `
<button
  className="base primary small disabled"
  onClick={[Function]}
>
  Testing
</button>
`;

exports[`src/ts/components/Button Rendering renders default correctly 1`] = `
<button
  className="base primary small "
  onClick={[Function]}
>
  Testing
</button>
`;

exports[`src/ts/components/Button Rendering size renders "large" correctly 1`] = `
<button
  className="base primary large "
  onClick={[Function]}
>
  Testing
</button>
`;

exports[`src/ts/components/Button Rendering tier renders "secondary" correctly 1`] = `
<button
  className="base secondary small "
  onClick={[Function]}
>
  Testing
</button>
`;
